src/shared/components/query/GenesetsVolcanoSelector.tsx

```diff
@@ -0,0 +1,220 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import LabeledCheckbox from "../labeledCheckbox/LabeledCheckbox";
+import * as styles_any from './styles/styles.module.scss';
+import {action, ObservableMap, expr, toJS, computed, observable, autorun} from "mobx";
+import {observer, Observer} from "mobx-react";
+import LazyMobXTable from "shared/components/lazyMobXTable/LazyMobXTable";
+import {Geneset} from "../../api/generated/CBioPortalAPIInternal";
+import {IColumnFormatterData} from "../enhancedReactTable/IColumnFormatter";
+import {IColumnDefMap} from "../enhancedReactTable/IEnhancedReactTableProps";
+import {Td} from 'reactable';
+import {toPrecision} from "../../lib/FormatUtils";
+import { getHierarchyData } from "shared/lib/StoreUtils";
+import ReactSelect from "react-select";
+import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
+import { VictoryChart, VictoryScatter, VictoryTheme, VictorySelectionContainer,
+    VictoryAxis, VictoryLabel, VictoryLine } from 'victory';
+import {QueryStoreComponent} from "./QueryStore";
+
+const styles = styles_any as {
+    GenesetsVolcanoSelectorWindow: string,
+    selectButton: string,
+    selectionColumnHeader: string,
+    selectionColumnCell: string,
+};
+
+class GenesetsVolcanoTable extends LazyMobXTable<Geneset> {}
+
+export interface GenesetsVolcanoSelectorProps
+{
+    initialSelection: string[];
+    data: Geneset[]|undefined;
+    plotData: {x: number, y: number, fill: string}[]|undefined;
+    maxY: number|undefined;
+    onSelect: (map_genesets_selected:ObservableMap<boolean>) => void;
+}
+
+@observer
+export default class GenesetsVolcanoSelector extends QueryStoreComponent<GenesetsVolcanoSelectorProps, {plotData:{x: number, y: number, fill: string}[]}>
+{
+    readonly percentileOptions = [{label: '50%', value: '50'}, {label: '75%', value: '75'}, {label: '100%', value: '100'}];
+    constructor(props:GenesetsVolcanoSelectorProps)
+    {
+        super(props);
+        this.percentileChange = this.percentileChange.bind(this);
+        this.updateSelectionFromPlot = this.updateSelectionFromPlot.bind(this);
+    }
+    
+    percentileChange(val: {label: string, value: string} | null)
+    {
+        this.store.volcanoPlotSelectedPercentile = val || {label: '75%', value: '75'};
+    }
+    
+    updateSelectionFromPlot(points: any, bounds: any) {
```

Comment:
this function should be @action since it updates observable state (map_genesets_selected_volcano) and does so multiple times.  @action will allow batch updating. this is critical to performance because it prevents duplicate render cycles.
also, this should have a unit test that makes sure map_genesets_selected_volcano is update properly.